pkgs/tools/misc/expect/default.nix

```diff
@@ -6,22 +6,36 @@ tcl.mkTclDerivation rec {
 
   src = fetchurl {
     url = "mirror://sourceforge/expect/Expect/${version}/expect${version}.tar.gz";
-    sha256 = "0d1cp5hggjl93xwc8h1y6adbnrvpkk0ywkd00inz9ndxn21xm9s9";
+    hash = "sha256-Safag7C92fRtBKBN7sGcd2e7mjI+QMR4H4nK92C5LDQ=";
   };
 
   patches = [
     (fetchpatch {
       url = "https://raw.githubusercontent.com/buildroot/buildroot/c05e6aa361a4049eabd8b21eb64a34899ef83fc7/package/expect/0001-enable-cross-compilation.patch";
-      sha256 = "1jwx2l1slidvcpahxbyqs942l81jd62rzbxliyd9lwysk38c8b6b";
+      hash = "sha256-yyzE0Jjac5qaj7Svn4VpMiAqSNLYrw7VZbtFqgMVncs=";
     })
     (substituteAll {
       src = ./fix-cross-compilation.patch;
       tcl = "${buildPackages.tcl}/bin/tclsh";
     })
+    # The following patches fix compilation with clang 15+
+    (fetchpatch {
+      url = "https://sourceforge.net/p/expect/patches/24/attachment/0001-Add-prototype-to-function-definitions.patch";
+      hash = "sha256-X2Vv6VVM3KjmBHo2ukVWe5YTVXRmqe//Kw2kr73OpZs=";
+    })
+    (fetchpatch {
+      url = "https://sourceforge.net/p/expect/patches/_discuss/thread/b813ca9895/6759/attachment/expect-configure-c99.patch";
+      hash = "sha256-PxQQ9roWgVXUoCMxkXEgu+it26ES/JuzHF6oML/nk54=";
+    })
   ];
 
   postPatch = ''
     sed -i "s,/bin/stty,$(type -p stty),g" configure.in
+    # `ioctl` requires including `sys/ioctl.h` on Darwin, which is safe to do unconditionally on Linux.
+    sed -i "/#ifdef HAVE_TERMIOS/{n;n;g;d}" exp_win.c
+  '' + lib.optionalString stdenv.isDarwin ''
+    # Darwin puts `openpty` in `util.h` not `pty.h` (unlike Linux).
+    sed -i "s/#include <pty.h>/#include <util.h>/" pty_termios.c
   '';
```

Comment:
Shouldn't those also be patches? This would prevent silent breakages on updates and is probably a lot simpler especially with regards to the sed in line 35.

Reply:
Sure, I’ll push an update that does this via a patch instead.

Reply:
I’ve pushed an updated PR using a patch.

Reply:
Thanks ❤️